### PR TITLE
ref(metrics): Move aggregator into server crate

### PR DESCRIPTION
**Pull request by Dav1dde**
Moves the metrics services we have into the server module. This is preparation to log outcomes for dropped messages, which needs access to the metric outcomes.

Contains no functional changes:
- Moves config into `relay-config` module.
- Moves the services and metrics into `relay-server`.

#skip-changelog

### CODE REVIEW

File: relay-config/src/config.rs [hunk 1]
```diff
@@ -1210,6 +1210,117 @@ impl Default for Outcomes {
     }
 }
 
+/// Parameters used for metric aggregation.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(default)]
+pub struct AggregatorServiceConfig {
```
Comment (jjbayer) [inline relay-config/src/config.rs]: While we're at it, should we move the newly added types to a submodule?

File: relay-server/src/services/router.rs
Comment (jjbayer) [inline relay-server/src/services/router.rs:1]: This was previously `relay_metrics::router`, should we now prefix it as `services::metrics_router`? Or better yet, `services::metrics::router`?

**Review by jjbayer — APPROVED** [review]
If it's just about making outcomes accessible in metrics, we could probably also use some inversion of control pattern and pass something like a `track_outcome` callback as an argument or a type parameter to the service.

**Comment by Dav1dde** [comment]
@jjbayer wanna do this in a followup and pass `MetricOutcomes` to the service, is that what you meant?

**Comment by jjbayer** [comment]
> wanna do this in a followup and pass `MetricOutcomes` to the service, is that what you meant?

I just wanted to point out that there's other ways to solve the dependency problem, if we want to keep the aggregator service in the metrics crate. But I'm fine with moving it into `relay-server`.

**Pull request merged**